docs/proposals/organizations.md

```diff
@@ -0,0 +1,118 @@
+     I'm in favor of keeping the current structure, but allowing deeper nesting of subresources and having named subresources.
+
+ 2.  Should we allow projects to be owned by Users or only allow ownership by Organizations?
+
+     Allowing Users to own projects eliminates the creation of another "linked" resource that needs to be protected and maintained.
+     Having a single entity allowed to own projects limits some duplication of things like quota.
+
+ 3.  Can we punt on adding users as organization members?
+
+     Organization-admins should not be able to force users into an Org.  Users should not be able to join any Org they want.  I'd like to get the structure correct and punt on the ability to add members.  It would require some sort of invitation/request/approval flow.
+
+ 4.  Can we punt on project transfering?
+
+     An org-admin shouldn't be able to take a project from someone else and they shouldn't be allowed to orphan their own project.  That means we'd need an offer/request/approval flow.
+
+
+
+## Organizations
+An Organization is a cluster scoped API object that references Users and has subresources for accessing and modifying Organization scoped OrgGroups.
+```
+// Organization contains the set of Users that are members of the Organization.
+type Organization struct {
+	kapi.TypeMeta
+	kapi.ObjectMeta
+
+	// AdminUsers is a list of Users that can administer this Organization.  By default a policy rule exists
+	// that allows all Users to inspect and modify OrgGroups of Organizations that they are an Admin for.
+	// Admins may NOT modify Organization membership.
+	AdminUsers util.StringSet
+	// AdminGroups is a list of Groups that can administer this Organization.  By default a policy rule exists
+	// that allows all Users to inspect and modify OrgGroups of Organizations that they are an Admin for.
+	// Admins may NOT modify Organization membership.
+	AdminGroups util.StringSet
+
+	// MemberUsers is a list of Users that are members of the Organization.  Only members may be assigned to OrgGroups.
+	MemberUsers util.StringSet
+	// MemberGroups is a list of Groups whose constituent users are considered members of the Organization.  Only members may be assigned to OrgGroups.
+	MemberGroups util.StringSet
+}
+```
+
+The Organization has a list of admins.  Admins have the power to:
+ 1.  Reserve quota for particular projects.
+ 2.  Accept ownership of a project.
+ 3.  Offer ownership of a project.
+ 4.  Remove (but not add) group members.
+ 5.  Create and modify organization groups.
+ 6.  Modify RoleBindings in projects the organization owns.  (This prevents accidentally orphaning a project.)
+
+The Organization has a list of members.  Only users from this list may be assigned to OrgGroups.  This prevents an overzealous organization admin from creating OrgGroups made up of non-members.
```

Comment:
Will we be able to determine if a user is member of an organization (and any orggroup) just by looking at the user object?